### DISCUSSION TITLE
How do you authenticate if you are using an external OIDC identity provider such as Azure AD?

### DISCUSSION THREAD

**Discussion by IsmaelJCarlo** [thycotic-ps/thycotic.secretserver#272] [Q&A : Module] [answered]
Do you have an example of how you would authenticate if you are using an external OIDC identity provider such as Azure AD with SecretServer cloud?

**Comment by jagger** [comment, marked as answer]
Thanks for raising this. The PowerShell module authenticates against Secret Servers existing /oauth2/token endpoint and cannot establish a session via an identity provider that Secret Server itself does not already accept on that endpoint. Adding native Azure AD / external OIDC auth would require server-side changes in Secret Server (to expose an OIDC-aware token endpoint that the module could call). It is not something we can implement in this module alone — please raise it with Delinea Support / the Secret Server roadmap so the platform side can be considered.
